bindings/python/src/metadata.rs

```diff
@@ -91,11 +91,40 @@ impl Metadata {
     pub fn mode(&self) -> EntryMode {
         EntryMode(self.0.mode())
     }
+
     /// Last modified time
     #[getter]
     pub fn last_modified(&self) -> Option<DateTime<Utc>> {
         self.0.last_modified()
     }
+    pub fn __repr__(&self) -> String {
+        let last_modified_str = match self.0.last_modified() {
+            Some(dt) => {
+                let rfc3339 = dt.to_rfc3339_opts(SecondsFormat::Micros, false);
+                if let Some(pos) = rfc3339.find('.') {
+                    let (base, tz) = rfc3339.split_at(pos + 1);
+                    let tz_pos = tz.find('+').or_else(|| tz.find('-')).unwrap_or(tz.len());
+                    let (micros, tz_suffix) = tz.split_at(tz_pos);
+
+                    let micros = format!("{:0<6}", micros);
+                    format!("{}{}{}", base, micros, tz_suffix)
+                } else {
+                    rfc3339
+                }
+            }
+            None => "None".to_string(),
+        };
+
+        format!(
+            "Metadata(content_length={}, content_md5={}, content_type={}, etag={}, mode={}, last_modified={})",
```

Comment:
I believe `mode` is the most important metadata, and `content_md5` can be removed since we don't encourage users to rely on it.

So the final list will be:

- mode
- content_length
- content_type
- last_modified
- etag

Reply:
copy